### PR TITLE
MOB-588: Added default page when there is not internet connection

### PR DESCRIPTION
**Pull request by m-revetria**
Fixes:
* [MOB-588: iOS 10 - IDP sign in screen frozen when network signal lost](https://idmeinc.atlassian.net/browse/MOB-588)
[MOB-589: iOS10- IDP sign in screen frozen when network signal lost and reconnected](https://idmeinc.atlassian.net/browse/MOB-589)

## Description
Added default page that is shown in case of error when the device is not connected to internet. This default page has a retry button that will reload the last request. This gives the user the ability to retry when the device connects again.

## Changes proposed in this request:
* Added a loading indicator in the web view. This indicator is shown the first time the web view is loading a request and it is dismissed after that request has finished.
* Refactor around WebView delegates implementation. Moved that logic away from `IDmeWebVerify` class and encapsulated in a the new class `IDmeAuthenticationDelegate`.
* Refactor `IDmeWebVerify` to get rid of multiple instance variables.
* Renamed some classes in order to let all classes' name start with prefix `IDme` to prevent name collision issues

image

## Todos
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by sdownie — APPROVED** [review]

File: ID.me WebVerify SDK/IDmeWebVerify.m [hunk 1]
```diff
@@ -7,13 +7,18 @@
...
-#import <WebKit/WebKit.h>
 #import "IDmeWebVerifyKeychainData.h"
-#import "ConnectionDelegate.h"
+#import "IDmeWebView.h"
+
+#import <WebKit/WebKit.h>
 
 /// API Constants (Production)
-#define IDME_WEB_VERIFY_BASE_URL                        @"https://api.id.me/"
+#define IDME_WEB_VERIFY_BASE_URL                        @"https://api.idmelabs.com/"
```
Comment (mats-claassen) [inline ID.me WebVerify SDK/IDmeWebVerify.m]: Please undo this

File: ID.me WebVerify SDK/IDmeWebVerify.m [hunk 2]
```diff
@@ -72,14 +73,16 @@ + (IDmeWebVerify *)sharedInstance {
...
+- (id)init {
     self = [super init];
     if (self) {
         _keychainData = [[IDmeWebVerifyKeychainData alloc] init];
-        connectionDelegate = [[ConnectionDelegate alloc] init];
         _showCancelButton = YES;
         isRefreshing = NO;
         pendingRefreshes = [[NSMutableDictionary alloc] init];
+        self.errorPageTitle = NSLocalizedString(@"Unavailable", @"IDme WebVerify SDK disconnected page title");
+        self.errorPageDescription = NSLocalizedString(@"ID.me requires a connection to the internet.", @"IDme WebVerify SDK disconnected page description");
```
Comment (mats-claassen) [inline ID.me WebVerify SDK/IDmeWebVerify.m]: @sdownie This changed to "ID.me Wallet requires an internet connection", right?

File: ID.me WebVerify SDK/IDmeWebView.m [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+    descriptionLabel.tag = DISCONNECTED_VIEW_DESCRIPTION_LABEL_TAG;
+    descriptionLabel.text = self.errorPageDescription;
+    descriptionLabel.textAlignment = NSTextAlignmentCenter;
+    descriptionLabel.textColor = textColor;
+    descriptionLabel.translatesAutoresizingMaskIntoConstraints = NO;
+
+    UIButton *retryButton = [[UIButton alloc] initWithFrame:CGRectMake(0, 0, 90, 30)];
+    retryButton.translatesAutoresizingMaskIntoConstraints = NO;
+    [retryButton addTarget:self action:@selector(retryButtonDidTouch:) forControlEvents:UIControlEventTouchUpInside];
+    [retryButton setTitle:@"Retry" forState:UIControlStateNormal];
```
Comment (mats-claassen) [inline ID.me WebVerify SDK/IDmeWebView.m]: Should texts be localized? Not sure if the texts were in the rest of the SDK

**Review by mats-claassen — CHANGES_REQUESTED** [review]

File: ID.me WebVerify SDK/IDmeWebVerify.m (hunk 2 above)
Comment (sdownie) [reply ID.me WebVerify SDK/IDmeWebVerify.m]: @mats-claassen Yes

**Review by sdownie — CHANGES_REQUESTED** [review]
Fix comments

File: ID.me WebVerify SDK/IDmeWebView.m (hunk 3 above)
Comment (sdownie) [reply ID.me WebVerify SDK/IDmeWebView.m]: @mats-claassen Don't need to be localized at this point

**Review by mats-claassen — APPROVED** [review]

**Review by sdownie — APPROVED** [review]

**Pull request merged**